docs/sources/flow/reference/components/pyroscope.scrape.md

```diff
@@ -305,6 +305,10 @@ by a service discovery component such as `discovery.kubernetes`. The special
 label `__address__` _must always_ be present and corresponds to the
 `<host>:<port>` that is used for the scrape request.
 
+The special label `service_name` is required and must always be present. If `service_name` is not explicitly provided, it
+will be inferred from the discovery meta labels. If the service name cannot be determined from the discovery meta
+labels, it will default to `unknown_service_name`.
```

Comment:
We also should keep in mind that as long as we don't support removal from the db, it's might be quite frustrating experience for users to have "unknown" service in the dropdown (same with high cardinality values, such as pod name) – we should be ready to answer this question. On the other hand, provided that we fetch labels only from the head block, the "unknown" app will eventually disappear from the list once the user configuration is fixed.

Reply:
just trying to understand what this will look like in the UI:
<img width="505" alt="image" src="https://github.com/grafana/agent/assets/23323466/e5003cbe-dc51-4f06-b861-d477bf4c4e16">

cc @eh-am

Reply:
If it defaults to `unknown_service_name`, that's how it will look like in the UI, yes.

Differently from Anton, I feel like it's fine to have `unknown` or even fail to scrape altogether. IMO it's better to fail than trying to be too clever. When one is first setting up it's expected to have a few hiccups. As long as we document (and have proper error messages) it should be fine.

>On the other hand, provided that we fetch labels only from the head block, the "unknown" app will eventually disappear from the list once the user configuration is fixed.
But my understanding is that this is going to change sometime in the future, no?

Reply:
> Differently from Anton, I feel like it's fine to have unknown or even fail to scrape altogether. IMO it's better to fail than trying to be too clever. When one is first setting up it's expected to have a few hiccups. As long as we document (and have proper error messages) it should be fine.

Just to clarify: I think that the label must be mandatory and explicit in both push and pull modes in the same way it is implemented in Pyroscope, and a missing service name should be threat as a misconfiguration. Profiles without the label should be discarded, and a corresponding message should be logged.

OTOH, I don't know how that works for ebpf.

Reply:
If it's not set it still seems like it would be easier to handle in the UI if there was some value that _means_ that it was not set (as opposed to not setting the label at all) wdyt @eh-am? would you rather have no `service_name` label or `service_name: undefined`

Reply:
after discussing further in slack we decided that `service_name:undefined` may be better for the frontend, but will make querying significantly slower (compared to having an explicit service name that querying can rely on).